3]. Competitive Programming/05]. HackerEarth/1]. Practice/5]. Code Monk Series/1] Arrays & Strings/Cyclic shift.cpp

```diff
@@ -0,0 +1,348 @@
+
+
+/* 
+
+    Solution by Rahul Surana
+
+    ***********************************************************
+
+A large binary number is represented by a string  of size A and N comprises of 0s and 1s. 
+You must perform a cyclic shift on this string. The cyclic shift operation is defined as follows:
+
+If the string A is [a0,a1,a2,...,a(n-1)], then after performing one cyclic shift, the string becomes [a1,a2,a3,...,a0].
+
+You performed the shift infinite number of times and each time you recorded the value of the binary number represented by the string. 
+The maximum binary number formed after performing (possibly 0) the operation is B. 
+Your task is to determine the number of cyclic shifts 
+that can be performed such that the value represented by the string A will be equal to B for the kth time.
+
+Input format
+
+First line: A single integer T denoting the number of test cases
+For each test case:
+First line: Two space-separated integers N and K 
+Second line: A denoting the string
+
+    ***********************************************************
+
+*/
+
+#include <bits/stdc++.h>
```

Comment:
check this code again
![image](https://user-images.githubusercontent.com/60146338/126860732-0c9f2620-5593-4758-a170-bfe49bba909b.png)